Revert "Set RPATH for installed binaries"

mmuetzel opened the pull request:
This reverts commit 222b7ee17b9b2ce6543b5165171c74f7b641efb4.

See #599 and https://gitlab.com/petsc/petsc/-/merge_requests/7104#note_1699384612

mmuetzel committed:
Revert "Set RPATH for installed binaries"

This reverts commit 222b7ee17b9b2ce6543b5165171c74f7b641efb4.

mmuetzel committed:
SuiteSparsePolicy.cmake: Fix copy-paste error in comment.

haampie commented on GraphBLAS/cmake_modules/SuiteSparsePolicy.cmake:
```diff
@@ -155,7 +155,7 @@ endif ( )
 list ( FIND CMAKE_BUILD_RPATH ${CMAKE_BINARY_DIR} _idx )
 if ( _idx LESS 0 )
-    # not yet included in CMAKE_INSTALL_RPATH
+    # not yet included in CMAKE_BUILD_RPATH
```
Notice that CMake always sets rpaths during the build, so that just-built executables can be run. I might be missing something, but I think this bit may be redundant.

It's on install that they are cleared by default.

mmuetzel replied:
Checking the commit history, the CMAKE_BUILD_RPATH was set since the very first version of `SuiteSparsePolicy.cmake` (and I never thought about it).

But iiuc, you are right: It should not be necessary to set it.

For a test, I've just removed it. Let's see if something breaks without it in CI.

haampie replied:
Could be that it's CMake < 3 remnants, or something

mmuetzel replied:
The CI passed all tests without issues. 👍

mmuetzel committed:
SuiteSparsePolicy.cmake: Don't append to CMAKE_BUILD_RPATH

On platforms that support it, the rpath should be set by default for
libraries before installation.

DrTimothyAldenDavis commented:
I can't recall.  It might date back to when cmake was first used in Mongoose or GraphBLAS.  I'm happy to remove it.
